Lib/glyphsLib/builder/glyph.py

```diff
@@ -256,6 +257,41 @@ def to_glyphs_glyph(self, ufo_glyph, ufo_layer, master):  # noqa: C901
     self.to_glyphs_paths(ufo_glyph, layer)
     self.to_glyphs_components(ufo_glyph, layer)
     self.to_glyphs_glyph_anchors(ufo_glyph, layer)
+    self.to_glyphs_glyph_height_and_vertical_origin(ufo_glyph, master, layer)
+
+
+def to_ufo_glyph_height_and_vertical_origin(self, ufo_glyph, layer):
+    # implentation based on:
+    # https://github.com/googlefonts/glyphsLib/issues/557#issuecomment-667074856
+    if not self.is_vertical:
+        return
+
+    ascender, descender = _vert_typesetting_metrics(layer.master)
+
+    if layer.vertWidth:
+        ufo_glyph.height = layer.vertWidth
+    else:
+        ufo_glyph.height = ascender - descender
+
+    if layer.vertOrigin:
+        ufo_glyph.verticalOrigin = ascender - layer.vertOrigin
+    else:
+        ufo_glyph.verticalOrigin = ascender
+
+
+def _vert_typesetting_metrics(master):
```

Comment:
this method is really only getting the typo ascender and descender, maybe name it `_get_typo_ascender_descender`?